src/jsi/jsi_class.hpp

```diff
@@ -125,10 +125,21 @@ inline void ObjectSetPrototypeOf(JsiEnv env, const fbjsi::Value& target, const f
     obj.getPropertyAsFunction(env, "setPrototypeOf").callWithThis(env, obj, target, proto);
 }
 
+// Cache various objects we fetch from the runtime which are hot paths
+// during object creation
+static std::optional<fbjsi::Function> s_object;
+static std::optional<fbjsi::Function> s_object_create;
+
 inline JsiObj ObjectCreate(JsiEnv env, const fbjsi::Object& proto)
 {
-    auto obj = js::globalType(env, "Object");
-    return env(obj.getPropertyAsFunction(env, "create").callWithThis(env, obj, proto)).asObject();
+    if (REALM_UNLIKELY(!s_object)) {
+        s_object = js::globalType(env, "Object");
+    }
+    if (REALM_UNLIKELY(!s_object_create)) {
+        s_object_create = (*s_object).getPropertyAsFunction(env, "create");
+    }
+
+    return env((*s_object_create).callWithThis(env, *s_object, proto)).asObject();
```

Comment:
```suggestion
    return env(s_object_create->callWithThis(env, *s_object, proto)).asObject();
```

FYI, (almost) any time you do `(*x).y` you can instead do `x->y`. The only exceptions are when they are overloaded to do different things (bad idea) or `*x` is provided but `x->` isn't (sometimes a good idea). Neither issue applies to `std::optional` though.

Out of curiousity is there any perf difference between this and `env(s_object_create->call(env, proto)).asObject()` (that is, stop attaching `Object` as `this`)? I wasn't sure if it was safe to call the static methods on `Object` without a bound this and I wasn't able to test this while I was writing it, so I went for the conservative option of always doing the equivalent of `Object.foo()` rather than `const foo = Object.foo; foo()`. But it seems somewhat unlikely that it would matter, and I could imagine it being faster to not pass the `this` argument. IIRC there were several places where I did something like this. If it makes any difference it may be worth searching for `callWithThis` calls and seeing how many can be replaced with just `call`. (In a separate PR)

Reply:
> FYI, (almost) any time you do (*x).y you can instead do x->y
Ah, of course, obvious in hindsight haha.

> Out of curiousity is there any perf difference between this and env(s_object_create->call(env, proto)).asObject()
Will take a look at this, thanks!

Reply:
>Out of curiousity is there any perf difference between this and env(s_object_create->call(env, proto)).asObject()

This does seem to yield a small (<5%) performance increase, good suggestion!

Reply:
Warrants more in depth testing probably as I just did a few runs and eyeballed the results, but yeah makes sense!